### PR TITLE
Api Explorer: Replace Sidebar with Accordion

### PR DESCRIPTION
**Pull request by josephaxisa**
TODO:
- [x] Fix styling

Please do not merge this before the styling branch (currently this PR's base branch) is merged.

### CODE REVIEW

**Comment by miketorosian** [comment]
@josephaxisa - Is it possible to add tooltips to truncated items? I think we'd still want users to be able to see the full name there. I don't think tooltips are necessary on all, just truncated ones if that's doable.

**Comment by jkaster** [comment]
> @josephaxisa - Is it possible to add tooltips to truncated items? I think we'd still want users to be able to see the full name there. I don't think tooltips are necessary on all, just truncated ones if that's doable.

That's definitely a good idea @miketorosian and should be feasible to do.

**Comment by miketorosian** [comment]
@jkaster and @josephaxisa - I did some cleanup and pushed. The nav nearly perfectly matches what it used to be now. I did notice in ExplorerStyle.tsx a few bits of CSS that I don't believe I added so if they are not needed we can probably remove for further cleanup:

`  .codeMarker {
    background: yellow;
    position: absolute;
  }

  .hi {
    background: yellow;
  }`

Aside from that, there were a bunch of unused styles in this file from my first round of styling work from way back so I removed all of that since it's no longer needed.

File: packages/api-explorer/src/components/SideNav/SideNavMethods.tsx [hunk 1]
```diff
@@ -40,33 +49,111 @@ interface MethodsProps {
...
+    }
+
+    &.active {
+      ${ApixHeading} {
+        font-weight: ${({ theme }) => theme.fontWeights.semiBold};
+      }
+    }
+  }
+`
+
```
Comment (josephaxisa) [inline packages/api-explorer/src/components/SideNav/SideNavMethods.tsx:110]: These should live in `ExplorerStyle.tsx`

**Comment by josephaxisa** [comment]
> ` .codeMarker {
> background: yellow;
> position: absolute;
> }
> 
> .hi {
> background: yellow;
> }`
> 
> Aside from that, t

Unfortunately when I highlight html using `codeMarker` instead of `hi` the below happens.

Main content area:
main content area highlighting

Sidenav:
Sidenav highlighting

It's as if the text is overlapping. If we get that fixed we'll only need one class for highlighting.

File: packages/api-explorer/src/ApiExplorer.tsx [hunk 2]
```diff
@@ -89,7 +89,10 @@ const ApiExplorer: FC<ApiExplorerProps> = ({ specs, tryItCallback }) => {
               isOpen={isSideNavOpen}
             />
           </SideNavDivider>
-          <Box className={isSideNavOpen ? 'main doc open' : 'main doc'}>
+          <Box
+            className={isSideNavOpen ? 'main doc open' : 'main doc'}
```
[the remark this answers is not in the document]
Comment (miketorosian) [reply packages/api-explorer/src/ApiExplorer.tsx:93]: @josephaxisa - That is leftover from the implementation of the collapsible sidebar and definitely is outside of the scope of what I am able to contribute to. If that's something you all want to tackle I have no objections but that goes beyond the scope of what I can help with.

File: packages/api-explorer/src/components/ExplorerStyle/ExplorerStyle.tsx [hunk 3]
```diff
@@ -116,56 +98,16 @@ export const PageLayout = styled.div<SideNavStyleProps>`
...
-  }
-
   .doc {
     grid-area: main;
   }
 `
 
 const reset = css`
   html {
-    color: #4c535b;
```
[the remark this answers is not in the document]
Comment (miketorosian) [reply packages/api-explorer/src/components/ExplorerStyle/ExplorerStyle.tsx:166]: @josephaxisa - The CSS reset just helps us get a good baseline but it's possible we don't even need this anymore. @lukelooker do you feel this reset is even necessary?

File: packages/api-explorer/src/components/SideNav/SideNavMethods.tsx (hunk 1 above)
Comment (miketorosian) [reply packages/api-explorer/src/components/SideNav/SideNavMethods.tsx:110]: @josephaxisa - There was some functionality here like `isOpen` that I wasn't sure how to move into ExplorerStyle.tsx if you can give me clarity on what I would need to bring over into ExplorerStyle.tsx to keep everything from breaking I am happy to clean this up further.

**Comment by miketorosian** [comment]
@josephaxisa - I am still not clear on what those two CSS pieces are doing for highlighting, I honestly don't recall ever adding those so I am not really sure why they are there or how they are supposed to be used. Maybe @jkaster has more context?

**Comment by miketorosian** [comment]
> @miketorosian I pulled your changes but I am still seeing the padding issue between the accordions on my side. I also made a few minor comments below.

Screen Shot 2020-07-08 at 8 20 03 AM

@josephaxisa - That's odd, this is what I see. Maybe try clearing your cache and refreshing to ensure you are seeing the latest? I'm not sure why we'd be seeing differences otherwise.

**Comment by josephaxisa** [comment]
> @josephaxisa - I am still not clear on what those two CSS pieces are doing for highlighting, I honestly don't recall ever adding those so I am not really sure why they are there or how they are supposed to be used. Maybe @jkaster has more context? As far as the problem you are seeing, I believe that is related to the use of `position: absolute` there. If you take that off, I bet it fixes that.

I was the one that introduced those two classes. `codeMarker` is for highlighting code in the editor and `hi` is for highlighting html. Unless we find a solution that fits both we're going to have to keep both.

**Comment by miketorosian** [comment]
> > @josephaxisa - I am still not clear on what those two CSS pieces are doing for highlighting, I honestly don't recall ever adding those so I am not really sure why they are there or how they are supposed to be used. Maybe @jkaster has more context? As far as the problem you are seeing, I believe that is related to the use of `position: absolute` there. If you take that off, I bet it fixes that.
> 
> I was the one that introduced those two classes. `codeMarker` is for highlighting code in the editor and `hi` is for highlighting html. Unless we find a solution that fits both we're going to have to keep both.

I think we can just use one, I believe the `position: absolute` is what's giving you issues because it's taking it out of the normal flow.

**Comment by josephaxisa** [comment]
> I think we can just use one, I believe the `position: absolute` is what's giving you issues because it's taking it out of the normal flow.

Removing the absolute messes it up inside the editors and vice versa. We can just leave it as is for now. I can rename `codeMarker` to `hiCode` for consistency.

**Comment by miketorosian** [comment]
> > I think we can just use one, I believe the `position: absolute` is what's giving you issues because it's taking it out of the normal flow.
> 
> Removing the absolute messes it up inside the editors and vice versa. We can just leave it as is for now. I can rename `codeMarker` to `hiCode` for consistency.

Screen Shot 2020-07-08 at 9 42 24 AM

@josephaxisa - I am not able to duplicate the highlighting issues you are experiencing on your end. On my screen, it is working without cutting off any text.

**Comment by miketorosian** [comment]
@josephaxisa it was suggested that in TagScene.tsx we use ButtonToggle instead so that the transparent button can be eliminated and it can use the styling for toggles instead. I added some styles for DocMethodSummmary in the meantime.

**Comment by josephaxisa** [comment]
Let's get this reviewed and merged. Additional action items such as the filtering behavior in method tag pages will be addressed in a separate PR.

**Pull request merged**